### ISSUE TITLE
After Authenticating, getting an java.lang.IllegalStateException: Not connected. Call connect() and wait for onConnected() to be called.

### ISSUE THREAD

**Issue by pletoss** [playgameservices/play-games-plugin-for-unity#781]
Hi,

On a couple of devices we get this error right after trying to log in ( Authenticate() ):

We assume that it only happends to users that don't have a Google Plus account yet or have never interacted with Google Play Games.

Infos:

I'm using GooglePlayGamesPlugin-0.9.26.unitypackage.

A dumpsys package for com.google.android.play.games shows this:

 Package [com.google.android.play.games] (4336b7f0):
    userId=10135 gids=[]
    pkg=Package{42c5e8e0 com.google.android.play.games}
    codePath=/data/app/com.google.android.play.games-6.apk
    resourcePath=/data/app/com.google.android.play.games-6.apk
    nativeLibraryPath=/data/app-lib/com.google.android.play.games-6
    versionCode=33580036 targetSdk=23
    versionName=3.3.58 (2285964-036)
    applicationInfo=ApplicationInfo{42c5e990 com.google.android.play.games}

Here's a stacktrace:

```
I/Unity   ( 6342):  [Play Games Plugin DLL] 10/26/15 11:22:47 +01:00 DEBUG: Starting Auth Transition. Op: SIGN_IN status: VALID
I/Unity   ( 6342):  
I/Unity   ( 6342): (Filename: ./artifacts/AndroidManagedGenerated/UnityEngineDebug.cpp Line: 56)
I/Unity   ( 6342): 
I/dalvikvm( 6169): Could not find method android.widget.TextView.setElegantTextHeight, referenced from method com.google.android.gms.games.ui.popup.BasePopup.loadView
W/dalvikvm( 6169): VFY: unable to resolve virtual method 9087: Landroid/widget/TextView;.setElegantTextHeight (Z)V
D/dalvikvm( 6169): VFY: replacing opcode 0x6e at 0x0071
D/SensorService(  807):   8.6 0.2 4.8
D/AndroidRuntime( 6342): Shutting down VM
W/dalvikvm( 6342): threadid=1: thread exiting with uncaught exception (group=0x41d8ada0)
D/SDK-EA-Android( 6342): Default thread exception handler caught something on thread main : 
D/SDK-EA-Android( 6342): java.lang.IllegalStateException: Not connected. Call connect() and wait for onConnected() to be called.
D/dalvikvm( 6342): Trying to load lib /data/app-lib/net.mantisshrimp.warlords-9/libgpg.so 0x0
W/dalvikvm( 6342): Shared lib '/data/app-lib/net.mantisshrimp.warlords-9/libgpg.so' already opened by CL 0x4278b3c0; can't open in 0x0
D/dalvikvm( 6342): Trying to load lib /data/app-lib/net.mantisshrimp.warlords-9/libgpg.so 0x4278b3c0
D/dalvikvm( 6342): Shared lib '/data/app-lib/net.mantisshrimp.warlords-9/libgpg.so' already loaded in same CL 0x4278b3c0
D/SDK-EA-Android( 6342): Starting to send unsent payloads
D/SDK-EA-Android( 6342): UploadTask started
D/HockeyApp( 6342): Writing unhandled exception to: /data/data/net.mantisshrimp.warlords/files/7ad136fc-e183-4741-b61a-d25d905f668d.stacktrace
E/AndroidRuntime( 6342): FATAL EXCEPTION: main
E/AndroidRuntime( 6342): Process: net.mantisshrimp.warlords, PID: 6342
E/AndroidRuntime( 6342): java.lang.Error: FATAL EXCEPTION [main]
E/AndroidRuntime( 6342): Unity version     : 4.6.6f2
E/AndroidRuntime( 6342): Device model      : samsung GT-I9301I
E/AndroidRuntime( 6342): Device fingerprint: samsung/s3ve3gxx/s3ve3g:4.4.2/KOT49H/I9301IXCUAOF1:user/release-keys
E/AndroidRuntime( 6342): 
E/AndroidRuntime( 6342): Caused by: java.lang.IllegalStateException: Not connected. Call connect() and wait for onConnected() to be called.
E/AndroidRuntime( 6342): 	at com.google.android.gms.common.internal.zzk.zzfc(Unknown Source)
E/AndroidRuntime( 6342): 	at com.google.android.gms.common.internal.zzk.zzjb(Unknown Source)
E/AndroidRuntime( 6342): 	at com.google.android.gms.games.internal.GamesClientImpl.zzno(Unknown Source)
E/AndroidRuntime( 6342): 	at com.google.android.gms.games.internal.GamesClientImpl.onConnected(Unknown Source)
E/AndroidRuntime( 6342): 	at com.google.android.gms.common.internal.zzl.zzj(Unknown Source)
E/AndroidRuntime( 6342): 	at com.google.android.gms.common.internal.zzl.zzfe(Unknown Source)
E/AndroidRuntime( 6342): 	at com.google.android.gms.common.internal.zzk$zzf.zzje(Unknown Source)
E/AndroidRuntime( 6342): 	at com.google.android.gms.common.internal.zzk$zza.zzc(Unknown Source)
E/AndroidRuntime( 6342): 	at com.google.android.gms.common.internal.zzk$zza.zzi(Unknown Source)
E/AndroidRuntime( 6342): 	at com.google.android.gms.common.internal.zzk$zzc.zzjg(Unknown Source)
E/AndroidRuntime( 6342): 	at com.google.android.gms.common.internal.zzk$zzb.handleMessage(Unknown Source)
E/AndroidRuntime( 6342): 	at android.os.Handler.dispatchMessage(Handler.java:102)
E/AndroidRuntime( 6342): 	at android.os.Looper.loop(Looper.java:136)
E/AndroidRuntime( 6342): 	at android.app.ActivityThread.main(ActivityThread.java:5476)
E/AndroidRuntime( 6342): 	at java.lang.reflect.Method.invokeNative(Native Method)
E/AndroidRuntime( 6342): 	at java.lang.reflect.Method.invoke(Method.java:515)
E/AndroidRuntime( 6342): 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1283)
E/AndroidRuntime( 6342): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1099)
E/AndroidRuntime( 6342): 	at dalvik.system.NativeStart.main(Native Method)
```

**Comment by claywilkinson** [comment]
Thanks for the details.  What version of Play Services are you running?  Is it only experienced by players without a G+ account?

**Comment by claywilkinson** [comment]
Also please check which version of the play services library you are using.  You can check this in the Android SDK manager, in the "Extras" section.  Make sure you are using the latest.

**Comment by pletoss** [comment]
Hi, thanks for responding so quick and sorry for my delay. For some reason my github notification emails weren't working correctly. 

The SDK Manager says revision 27. But we're using the shipped version from  GooglePlayGamesPlugin-0.9.26.unitypackage of the google-play-services.jar, which is:

 <integer name="google_play_services_version">7095000</integer>
 
I can't say for sure if it only happends for users that don't have a G+ account, there was just some anecdotal evidence during our tests where it happened about 2-3 times. 

From our live users crash reports we couldn't gather that information, just that it crashed with the same stacktrace.

**Comment by claywilkinson** [comment]
I believe the problem is the version of play services client you application is built with. If you update the play services client library it will work as expected.
